src/lib.rs

```diff
@@ -69,6 +69,19 @@ pub mod contract {
         pub ticks: Vec<Tick>,
     }
 
+    #[derive(scale::Decode, Default, scale::Encode, Clone, Debug)]
+    #[cfg_attr(
+        feature = "std",
+        derive(scale_info::TypeInfo, ink::storage::traits::StorageLayout,)
+    )]
+    pub struct SwapParams {
+        pool_key: PoolKey,
+        x_to_y: bool,
+        amount: TokenAmount,
```

Comment:
Caller specifies `TokenAmount` only once. Then, the output amount of one hop is used as the input amount for another